src/render.tsx

```diff
@@ -1,18 +1,31 @@
 import { MarkdownChildren, MarkdownElement } from "./model";
-import { synchronousRenderFunctions } from "./renderFunctions";
+import {
+  asynchronousRenderFunctions,
+  synchronousRenderFunctions,
+} from "./renderFunctions";
 
 /** @internal */
-function renderNode(element: MarkdownChildren): string {
+function renderNodeSync(element: MarkdownChildren): string {
   const results = synchronousRenderFunctions
-    .map((renderFunction) => renderFunction(element, renderNode))
+    .map((renderFunction) => renderFunction(element, renderNodeSync))
     .filter((result) => result !== null) as string[];
   if (results.length === 0) {
     throw new Error("Invalid element");
   }
   return results[0];
 }
 
+/** @internal */
+function renderNodeAsync(element: MarkdownChildren): Promise<string> {
+  const results = asynchronousRenderFunctions
+    .map((renderFunction) => renderFunction(element, renderNodeAsync))
+    .filter((result) => result !== null);
+  return Promise.resolve(
+    results.length === 0 ? renderNodeSync(element) : (results[0] as string)
+  );
+}
+
```

Comment:
I think we can simplify this by renaming "synchronousRenderFunctions" to "renderFunctions" and just adding the new one to the list